### DISCUSSION TITLE
MAUI.Blazor, greatest invention ever

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/maui#17441] [General]
The way it works is having a platform specific webview contained in native window, while dotnet is running it talks to the webview to change it.

It means you can do : 
1. Cross origin api calls. If you do these in web app browser won't be your friend. But now because you are making these calls outside of the browser you are your own friend.
2. Web UI
3. Mix any language : C/C++ interop, other .NET languages, other language via compilation to javascript / WASM, dynamically load javascript / WASM module in browser, execute WASM directly inside dotnet because there is a WASM interpretor
4. Multi-threading. There's yet threads supported in Blazor if you require threads this is the way to use (in theory blazor webassembly is multi-threaded in .NET 8, or you can use javascript workers)
5. Tighter integration with operating system makes it a traditional App, but isolation possible via .NET assembly trust level & executing codes in webassembly virtual machine in .NET

**Comment by LuohuaRain** [comment]
Yes, I can't agree more!

**Comment by LuohuaRain** [comment]
But it would be better if Comet comes back!

**Reply by Xyncgas** [reply]
Comet is Xamarin components you don't get web UI, is it better
